emperor/format.py

```diff
@@ -94,18 +97,20 @@ def format_pcoa_to_js(header, coords, eigvals, pct_var, custom_axes=[],
     # write the values for all the spheres
     js_pcoa_string += '\nvar g_spherePositions = new Array();\n'
     for point, coord in zip(header, coords):
-        all_coords = ', '.join(["'P%d': %f" % (i+1,coord[i]) for i in range(number_of_axes)])
-        js_pcoa_string += ("g_spherePositions['%s'] = { 'name': '%s', 'color': "
-            "0, 'x': %f, 'y': %f, 'z': %f, %s };\n" % (point, point, coord[0],
-            coord[1],coord[2], all_coords))
+        all_coords = ', '.join(["'P%d': %f" % (i+1, coord[i]) for i in
+                               range(number_of_axes)])
+        js_pcoa_string += ("g_spherePositions.push({ 'name': '%s', 'color': "
```

Comment:
While the emphasis is on the js, this is operation is quadratic. Suggest doing the list/join pattern